sdk/src/gateway.ts

```diff
@@ -1,36 +1,96 @@
+import { ethers, AbiCoder } from "ethers";
+
 export type EvmAddress = string;
 
 type GatewayQuote = {
-    onramp_address: EvmAddress;
-    dust_threshold: number;
+    gatewayAddress: EvmAddress;
+    dustThreshold: number;
     satoshis: number;
     fee: number;
     gratuity: string;
-    bitcoin_address: string;
-    tx_proof_difficulty_factor: number;
+    bitcoinAddress: string;
+    txProofDifficultyFactor: number;
+
+    // TODO: add this to API or add here
+    strategyAddress: EvmAddress | null,
+};
+
+type GatewayCreateOrderRequest = {
+    gatewayAddress: EvmAddress,
+    strategyAddress: EvmAddress | null,
+    satsToConvertToEth: number,
+    userAddress: EvmAddress,
+    gatewayExtraData: string | null,
+    strategyExtraData: string | null,
+    satoshis: number,
 };
 
 type GatewayOrderResponse = {
-    onramp_address: EvmAddress;
-    token_address: EvmAddress;
+    gatewayAddress: EvmAddress;
+    tokenAddress: EvmAddress;
     txid: string;
     status: boolean;
     timestamp: number;
     tokens: string;
     satoshis: number;
     fee: number;
-    tx_proof_difficulty_factor: number;
+    txProofDifficultyFactor: number;
+};
+
+type GatewayStartOrderResult = {
+    orderUuid: string,
+    opReturnHash: string,
 };
 
+export type GatewayParams = {
+    fromChain: 'Bitcoin', // Source chain
+    fromToken: 'BTC', // Source token
+    fromUserAddress: EvmAddress, // Source chain wallet address
```

Comment:
This is a `BitcoinAddress`, right?

Reply:
Or long-term, an enum of one or the other

Reply:
Right now this is a copy of the Swing params, we won't do anything with it really